mc-oblivious-ram/src/evictor/mod.rs

```diff
@@ -56,16 +142,125 @@ where
         stash_meta: &mut [A8Bytes<MetaSize>],
         branch: &mut BranchCheckout<ValueSize, Z>,
     ) {
-        branch.pack();
-        //Greedily place elements of the stash into the branch as close to the leaf as
-        // they can go.
-        for idx in 0..stash_data.len() {
-            branch.ct_insert(1.into(), &stash_data[idx], &mut stash_meta[idx]);
+        path_oram_eviction_strategy::<ValueSize, Z>(stash_data, stash_meta, branch);
+    }
+}
+
+/// Eviction algorithm defined in path oram. Packs the branch and greedily
+/// tries to evict everything from the stash into the checked out branch
+fn path_oram_eviction_strategy<ValueSize, Z>(
+    stash_data: &mut [A64Bytes<ValueSize>],
+    stash_meta: &mut [A8Bytes<MetaSize>],
+    branch: &mut BranchCheckout<ValueSize, Z>,
+) where
+    ValueSize: ArrayLength<u8> + PartialDiv<U8> + PartialDiv<U64>,
+    Z: Unsigned + Mul<ValueSize> + Mul<MetaSize>,
+    Prod<Z, ValueSize>: ArrayLength<u8> + PartialDiv<U8>,
+    Prod<Z, MetaSize>: ArrayLength<u8> + PartialDiv<U8>,
+{
+    branch.pack();
+    //Greedily place elements of the stash into the branch as close to the leaf as
+    // they can go.
+    for idx in 0..stash_data.len() {
+        branch.ct_insert(1.into(), &stash_data[idx], &mut stash_meta[idx]);
+    }
+}
+
+pub trait BranchSelector {
+    /// Returns the leaf index of the next branch to call evict from stash
+    /// to branch on.
+    fn get_next_branch_to_evict(&mut self) -> u64;
+
+    /// Returns the number of branches to call evict from stash to branch
+    /// on.
```

Comment:
:wrench: Suggest explicitly linking to the function
```suggestion
    /// Returns the leaf index of the next branch to call 
    /// [EvictionStrategy::evict_from_stash_to_branch] on.
    fn get_next_branch_to_evict(&mut self) -> u64;

    /// Returns the number of branches to call 
    /// [EvictionStrategy::evict_from_stash_to_branch] on.

```